src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1850,9 +1861,10 @@ private function registerValidatorMapping(ContainerBuilder $container, array $co
             $files['yaml' === $extension ? 'yml' : $extension][] = $path;
         };
 
-        if (ContainerBuilder::willBeAvailable('symfony/form', Form::class, ['symfony/framework-bundle', 'symfony/validator'])) {
-            $reflClass = new \ReflectionClass(Form::class);
-            $fileRecorder('xml', \dirname($reflClass->getFileName()).'/Resources/config/validation.xml');
+        if (!ContainerBuilder::willBeAvailable('symfony/form', Form::class, ['symfony/framework-bundle', 'symfony/validator'])) {
+            $container->removeDefinition('validator.form.attribute_metadata');
+        } elseif (!($r = new \ReflectionClass(Form::class))->getAttributes(Traverse::class) || !class_exists(ValidatorAttributeMetadataPass::class)) {
+            $fileRecorder('xml', \dirname($r->getFileName()).'/Resources/config/validation.xml');
```

Comment:
I suggest adding a comment that this is for `symfony/form < 7.4 or symfony/validator < 7.4`, to make it easier to clean this code when bumping the min version of those packages supported in FrameworkBundle (which will likely happen at some point during the 8.x lifecycle)